integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextmove/ServiceBusRestClient.java

```diff
@@ -0,0 +1,181 @@
+            return Optional.empty();
+        }
+
+        ServiceBusMessage.ServiceBusMessageBuilder sbmBuilder = ServiceBusMessage.builder();
+        String brokerPropertiesJson = response.getHeaders().getFirst("BrokerProperties");
+        JsonParser jsonParser = new JsonParser();
+        JsonObject brokerProperties = jsonParser.parse(brokerPropertiesJson).getAsJsonObject();
+        sbmBuilder.lockToken(brokerProperties.get("LockToken").getAsString())
+                .messageId(brokerProperties.get("MessageId").getAsString())
+                .sequenceNumber(brokerProperties.get("SequenceNumber").getAsString());
+
+        try {
+            Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
+            EduDocument eduDocument = unmarshaller.unmarshal(new StringSource(response.getBody()), EduDocument.class).getValue();
+            sbmBuilder.body(eduDocument);
+            log.debug(format("Received message on queue=%s with conversationId=%s", localQueuePath, eduDocument.getConversationId()));
+        } catch (JAXBException e) {
+            e.printStackTrace();
+        }
+
+        return Optional.of(sbmBuilder.build());
+    }
+
+    public void deleteMessage(ServiceBusMessage message) {
+        String resourceUri = format("https://%s.servicebus.windows.net/%s/messages/%s/%s",
+                props.getNextbest().getServiceBus().getNamespace(),
+                localQueuePath,
+                message.getMessageId(),
+                message.getLockToken());
+
+        String auth = createAuthorizationHeader(resourceUri);
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("Authorization", auth);
+        HttpEntity<Object> httpEntity = new HttpEntity<>(headers);
+
+        URI uri = convertToUri(resourceUri);
+        RestTemplate restTemplate = new RestTemplate();
+        ResponseEntity<String> response = restTemplate.exchange(uri, HttpMethod.DELETE, httpEntity, String.class);
+        if (!response.getStatusCode().is2xxSuccessful()) {
+            log.error("{} got response {}, message [conversationId={}] was not deleted",
+                    resourceUri, response.getStatusCode().toString(),
+                    message.getBody().getConversationId());
+        }
+    }
+
+    private URI convertToUri(String uriString) {
+        try {
+            return new URI(uriString);
+        } catch (URISyntaxException e) {
+            throw new MeldingsUtvekslingRuntimeException(format("Uri syntax error in: %s", uriString), e);
+        }
+    }
+
+    private String createAuthorizationHeader(String resourceUri) {
+
+        String urlEncoded;
+        try {
+            urlEncoded = URLEncoder.encode(resourceUri, "UTF-8");
+        } catch (UnsupportedEncodingException e) {
+            throw new MeldingsUtvekslingRuntimeException(e);
+        }
+
+        int expiry = Math.round(Instant.now().plusSeconds(20).toEpochMilli() / 1000);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Cast one of the operands of this division operation to a "float". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS2184)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this silly call to "Math.round" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS2185)